base/structure.py

```diff
@@ -1276,36 +1505,39 @@ def near_realoffset(self, offset):
         owner = self.owner
 
         # Start by getting our bounds.
-        minimum, maximum = idaapi.get_struc_first_offset(owner.ptr), idaapi.get_struc_last_offset(owner.ptr)
-        if (offset < minimum) or (offset >= maximum):
+        minimum, maximum = owner.realbounds
+        if not (minimum <= offset < maximum):
             cls = self.__class__
-            logging.info(u"{:s}({:#x}).members.near_realoffset({:+#x}) : Requested offset not within bounds {:#x}<->{:#x}. Trying anyways..".format('.'.join([__name__, cls.__name__]), owner.id, offset, minimum, maximum))
-
-        # Try and find the exact offset, because if we can..then we don't need
-        # to execute the rest of this function.
-        mem = idaapi.get_member(owner.ptr, offset)
-        if mem:
-            index = self.index(mem)
-            return self[index]
+            logging.warning(u"{:s}({:#x}).members.near_realoffset({:+#x}) : Requested offset not within bounds {:#x}<->{:#x}. Trying anyways..".format('.'.join([__name__, cls.__name__]), owner.id, offset, minimum, maximum))
```

Comment:
This would read better as "Requested offset is not within the bounds of the structure ({:#x}<>{:#x})".